File: converted/source2/source/zombie/audio/ObjectAmbientEmitters$ChunkData.java
package zombie.audio;

import java.util.HashMap;
import zombie.audio.ObjectAmbientEmitters.PerObjectLogic;
import zombie.iso.IsoObject;

public final class ObjectAmbientEmitters$ChunkData {
   final HashMap<IsoObject, PerObjectLogic> m_objects = new HashMap<>();

   public boolean hasObject(IsoObject var1) {
      return this.m_objects.containsKey(var1);
   }

   public void addObject(IsoObject var1, PerObjectLogic var2) {
      if (!this.m_objects.containsKey(var1)) {
         this.m_objects.put(var1, var2);
      }
   }

   public void removeObject(IsoObject var1) {
      this.m_objects.remove(var1);
   }

   public void reset() {
      this.m_objects.clear();
   }
}

File: ProjectZomboidSwitch/src/zombie/audio/ObjectAmbientEmitters$ChunkData.cpp
#include "zombie/audio/ObjectAmbientEmitters.h"

namespace zombie {
namespace audio {

bool ChunkData::hasObject(const IsoObject* obj) const {
    return m_objects.find(const_cast<IsoObject*>(obj)) != m_objects.end();
}

void ChunkData::addObject(IsoObject* obj, PerObjectLogic* logic) {
    m_objects.emplace(obj, logic);
}

void ChunkData::removeObject(IsoObject* obj) {
    m_objects.erase(obj);
}

void ChunkData::reset() {
    m_objects.clear();
}

} // namespace audio
} // namespace zombie
